### PR TITLE
Build recipes with channel_targets in cbc

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by chrisdembia** [comment]
Hey @isuruf I'm still interested in when this might get merged. Let me know if I can help.

**Comment by chrisdembia** [comment]
Thanks a bunch for working on this :)

**Comment by isuruf** [comment]
@conda-forge/staged-recipes, needs a review

File: .ci_support/build_all.py [hunk 1]
```diff
@@ -47,6 +87,20 @@ def build_all(recipes_dir, arch):
...
         exclusive_config_file=exclusive_config_file, channel_urls=channel_urls)
+    return config
+
+def build_folders(recipes_dir, folders, arch, channel_urls):
+
+    index_path = os.path.join(sys.exec_prefix, 'conda-bld')
+    try:
+        os.mkdir(index_path)
+    except FileExistsError:
+        pass
```
Comment (scopatz) [inline .ci_support/build_all.py]: I think that `os.makedirs(index_path, exist_ok=True)` is cleaner than the try-except here

File: .ci_support/build_all.py [hunk 2]
```diff
@@ -20,20 +25,55 @@ def get_host_platform():
...
+        old_comp_folders.extend(folders)
+    else:
+        for folder in folders:
+            built = False
+            cbc = os.path.join(recipes_dir, folder, "conda_build_config.yaml")
+            if os.path.exists(cbc):
+                with open(cbc, "r") as f:
+                    text = ''.join(f.readlines())
+                if 'channel_sources' in text:
+                    specific_config = safe_load(open(cbc))
```
Comment (scopatz) [inline .ci_support/build_all.py]: This file object isn't explicitly closed, right?  I would do:

```python
with open(cbc) as f:
    specific_config = safe_load(f)
```

**Comment by scopatz** [comment]
Thanks @isuruf! Looks good other than a couple of suggestions

**Comment by chrisdembia** [comment]
Thank you for reviewing this , @scopatz. I hope you’re doing well :). Have you been making any more fun videos for conferences?

**Comment by chrisdembia** [comment]
Thank you!

**Pull request merged**